// ng2-select/select/select.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ng2-select/select/off-click';
import * as i3 from 'ng2-select/select/select-pipes';
import * as i4 from 'ng2-select/select/select';
import * as i5 from '@angular/forms';
import * as i6 from '@angular/platform-browser';
const styles_SelectComponent:any[] = ['.ui-select-toggle[_ngcontent-%COMP%] {\n    position: relative;\n  }\n\n  \n  .ui-select-placeholder[_ngcontent-%COMP%] {\n    float: left;\n  }\n  \n  \n  .input-group[_ngcontent-%COMP%]    > .dropdown[_ngcontent-%COMP%] {\n    \n    position: static;\n  }\n  \n  .ui-select-match[_ngcontent-%COMP%]    > .btn[_ngcontent-%COMP%] {\n    \n    text-align: left !important;\n  }\n  \n  .ui-select-match[_ngcontent-%COMP%]    > .caret[_ngcontent-%COMP%] {\n    position: absolute;\n    top: 45%;\n    right: 15px;\n  }\n  \n  .ui-disabled[_ngcontent-%COMP%] {\n    background-color: #eceeef;\n    border-radius: 4px;\n    position: absolute;\n    width: 100%;\n    height: 100%;\n    z-index: 5;\n    opacity: 0.6;\n    top: 0;\n    left: 0;\n    cursor: not-allowed;\n  }\n  \n  .ui-select-choices[_ngcontent-%COMP%] {\n    width: 100%;\n    height: auto;\n    max-height: 200px;\n    overflow-x: hidden;\n    margin-top: 0;\n  }\n  \n  .ui-select-multiple[_ngcontent-%COMP%]   .ui-select-choices[_ngcontent-%COMP%] {\n    margin-top: 1px;\n  }\n  .ui-select-choices-row[_ngcontent-%COMP%] > a[_ngcontent-%COMP%] {\n      display: block;\n      padding: 3px 20px;\n      clear: both;\n      font-weight: 400;\n      line-height: 1.42857143;\n      color: #333;\n      white-space: nowrap;\n  }\n  .ui-select-choices-row.active[_ngcontent-%COMP%] > a[_ngcontent-%COMP%] {\n      color: #fff;\n      text-decoration: none;\n      outline: 0;\n      background-color: #428bca;\n  }\n  \n  .ui-select-multiple[_ngcontent-%COMP%] {\n    height: auto;\n    padding:3px 3px 0 3px;\n  }\n  \n  .ui-select-multiple[_ngcontent-%COMP%]   input.ui-select-search[_ngcontent-%COMP%] {\n    background-color: transparent !important; \n    border: none;\n    outline: none;\n    box-shadow: none;\n    height: 1.6666em;\n    padding: 0;\n    margin-bottom: 3px;\n    \n  }\n  .ui-select-match[_ngcontent-%COMP%]   .close[_ngcontent-%COMP%] {\n      font-size: 1.6em;\n      line-height: 0.75;\n  }\n  \n  .ui-select-multiple[_ngcontent-%COMP%]   .ui-select-match-item[_ngcontent-%COMP%] {\n    outline: 0;\n    margin: 0 3px 3px 0;\n  }\n  .ui-select-toggle[_ngcontent-%COMP%]    > .caret[_ngcontent-%COMP%] {\n      position: absolute;\n      height: 10px;\n      top: 50%;\n      right: 10px;\n      margin-top: -2px;\n  }'];
export const RenderType_SelectComponent:i0.RendererType2 = i0.ɵcrt({encapsulation:0,
    styles:styles_SelectComponent,data:{}});
function View_SelectComponent_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class',
      'ui-select-placeholder text-muted']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.placeholder;
        _ck(_v,1,0,currVal_0);
      });
}
function View_SelectComponent_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',[['class',
      'ui-select-match-text pull-left']],[[8,'innerHTML',1]],(null as any),(null as any),
      (null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'ui-select-allow-clear':0})],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'ui-select-match-text pull-left';
    const currVal_2:any = _ck(_v,2,0,(_co.allowClear && (_co.active.length > 0)));
    _ck(_v,1,0,currVal_1,currVal_2);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.sanitize(_co.active[0].text);
    _ck(_v,0,0,currVal_0);
  });
}
function View_SelectComponent_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'a',[['class',
      'btn btn-xs btn-link pull-right'],['style','margin-right: 10px; padding: 0;']],
      (null as any),[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.removeClick(_co.active[0],$event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n           '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'i',[['class','glyphicon glyphicon-remove']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        ']))],(null as any),(null as any));
}
function View_SelectComponent_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),17,'div',[['class',
      'ui-select-match']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),14,'span',[['class','btn btn-default btn-secondary form-control ui-select-toggle'],
          ['style','outline: 0;'],['tabindex','-1']],(null as any),[[(null as any),
          'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.matchClick($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_SelectComponent_3)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_SelectComponent_4)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),0,'i',[['class','dropdown-toggle pull-right']],
              (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),0,'i',[['class','caret pull-right']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,
          (null as any),View_SelectComponent_5)),i0.ɵdid(16384,(null as any),0,i1.NgIf,
          [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
          i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵted((null as any),['\n    ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = (_co.active.length <= 0);
        _ck(_v,5,0,currVal_0);
        const currVal_1:any = (_co.active.length > 0);
        _ck(_v,8,0,currVal_1);
        const currVal_2:any = (_co.allowClear && (_co.active.length > 0));
        _ck(_v,15,0,currVal_2);
      },(null as any));
}
function View_SelectComponent_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'input',[['autocomplete',
      'false'],['class','form-control ui-select-search'],['tabindex','-1'],['type',
      'text']],[[8,'disabled',0],[8,'placeholder',0]],[[(null as any),'keydown'],[(null as any),
      'keyup']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('keydown' === en)) {
      const pd_0:any = ((<any>_co.inputEvent($event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('keyup' === en)) {
      const pd_1:any = ((<any>_co.inputEvent($event,true)) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },(null as any),(null as any)))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.disabled;
    const currVal_1:any = i0.ɵinlineInterpolate(1,'',((_co.active.length <= 0)? _co.placeholder: ''),
        '');
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_SelectComponent_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),10,'li',[['role',
      'menuitem']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),7,'div',[['class','ui-select-choices-row']],[[2,'active',(null as any)]],
          [[(null as any),'mouseenter'],[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('mouseenter' === en)) {
              const pd_0:any = ((<any>_co.selectActive(_v.context.$implicit)) !== false);
              ad = (pd_0 && ad);
            }
            if (('click' === en)) {
              const pd_1:any = ((<any>_co.selectMatch(_v.context.$implicit,$event)) !== false);
              ad = (pd_1 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),4,'a',[['class','dropdown-item'],
          ['href','javascript:void(0)']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n              '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),[[8,'innerHTML',
          1]],(null as any),(null as any),(null as any),(null as any))),i0.ɵppd(2),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n          '])),(_l()(),i0.ɵted((null as any),['\n        ']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.isActive(_v.context.$implicit);
        _ck(_v,2,0,currVal_0);
        const currVal_1:any = _co.sanitize(i0.ɵunv(_v,6,0,_ck(_v,7,0,i0.ɵnov((<any>(<any>(<any>_v.parent).parent).parent),
            0),_v.context.$implicit.text,_co.inputValue)));
        _ck(_v,6,0,currVal_1);
      });
}
function View_SelectComponent_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'ul',[['class',
      'ui-select-choices dropdown-menu'],['role','menu']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_SelectComponent_8)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n      ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.options;
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
function View_SelectComponent_11(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'divider dropdown-divider']],(null as any),(null as any),(null as any),(null as any),
      (null as any)))],(null as any),(null as any));
}
function View_SelectComponent_12(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class',
      'ui-select-choices-row']],[[2,'active',(null as any)]],[[(null as any),'mouseenter'],
      [(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('mouseenter' === en)) {
      const pd_0:any = ((<any>_co.selectActive(_v.context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    if (('click' === en)) {
      const pd_1:any = ((<any>_co.selectMatch(_v.context.$implicit,$event)) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'active':0}),(_l()(),i0.ɵted((null as any),
      ['\n            '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'a',[['class',
      'dropdown-item'],['href','javascript:void(0)']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n              '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),[[8,'innerHTML',
          1]],(null as any),(null as any),(null as any),(null as any))),i0.ɵppd(2),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n          ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'ui-select-choices-row';
    const currVal_2:any = _ck(_v,2,0,_co.isActive(_v.context.$implicit));
    _ck(_v,1,0,currVal_1,currVal_2);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.isActive(_v.context.$implicit);
    _ck(_v,0,0,currVal_0);
    const currVal_3:any = _co.sanitize(i0.ɵunv(_v,6,0,_ck(_v,7,0,i0.ɵnov((<any>(<any>(<any>(<any>_v.parent).parent).parent).parent),
        0),_v.context.$implicit.text,_co.inputValue)));
    _ck(_v,6,0,currVal_3);
  });
}
function View_SelectComponent_10(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),10,'li',[['role',
      'menuitem']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_SelectComponent_11)),i0.ɵdid(16384,(null as any),
          0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'div',[['class','dropdown-header']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['',''])),(_l()(),i0.ɵted((null as any),['\n  \n          '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_SelectComponent_12)),i0.ɵdid(802816,
          (null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
          {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        ']))],
      (_ck,_v) => {
        const currVal_0:any = (_v.context.index > 0);
        _ck(_v,3,0,currVal_0);
        const currVal_2:any = _v.context.$implicit.children;
        _ck(_v,9,0,currVal_2);
      },(_ck,_v) => {
        const currVal_1:any = _v.context.$implicit.text;
        _ck(_v,6,0,currVal_1);
      });
}
function View_SelectComponent_9(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'ul',[['class',
      'ui-select-choices dropdown-menu'],['role','menu']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_SelectComponent_10)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n      ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.options;
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
function View_SelectComponent_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),19,'div',[['class',
      'ui-select-container dropdown open'],['tabindex','0']],(null as any),[[(null as any),
      'keyup'],[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>i0.ɵnov(_v,1).onClick($event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('keyup' === en)) {
      const pd_1:any = ((<any>_co.mainClick($event)) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(212992,(null as any),0,i2.OffClickDirective,
      ([] as any[]),{offClickHandler:[0,'offClickHandler']},(null as any)),(_l()(),
      i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      2,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,'ngClass']},(null as any)),
      i0.ɵpod({'ui-disabled':0}),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_SelectComponent_2)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_SelectComponent_6)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n     '])),(_l()(),
          i0.ɵted((null as any),['\n     '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_SelectComponent_7)),i0.ɵdid(16384,(null as any),
          0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n  \n      '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_SelectComponent_9)),i0.ɵdid(16384,(null as any),
          0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.clickedOutside;
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _ck(_v,5,0,_co.disabled);
    _ck(_v,4,0,currVal_1);
    const currVal_2:boolean = !_co.inputMode;
    _ck(_v,8,0,currVal_2);
    const currVal_3:any = _co.inputMode;
    _ck(_v,11,0,currVal_3);
    const currVal_4:any = (((_co.optionsOpened && _co.options) && (_co.options.length > 0)) && !_co.firstItemHasChildren);
    _ck(_v,15,0,currVal_4);
    const currVal_5:any = (((_co.optionsOpened && _co.options) && (_co.options.length > 0)) && _co.firstItemHasChildren);
    _ck(_v,18,0,currVal_5);
  },(null as any));
}
function View_SelectComponent_14(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),11,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      8,'span',[['class','ui-select-match-item btn btn-default btn-secondary btn-xs'],
          ['tabindex','-1'],['type','button']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'btn-default':0}),(_l()(),i0.ɵted((null as any),
      ['\n               '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'a',
      [['class','close'],['style','margin-left: 5px; padding: 0;']],(null as any),
      [[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.removeClick(_v.context.$implicit,$event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['×'])),(_l()(),
      i0.ɵted((null as any),['\n               '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),0,'span',([] as any[]),[[8,'innerHTML',1]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n           '])),
      (_l()(),i0.ɵted((null as any),['\n        ']))],(_ck,_v) => {
    const currVal_0:any = 'ui-select-match-item btn btn-default btn-secondary btn-xs';
    const currVal_1:any = _ck(_v,4,0,true);
    _ck(_v,3,0,currVal_0,currVal_1);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_2:any = _co.sanitize(_v.context.$implicit.text);
    _ck(_v,9,0,currVal_2);
  });
}
function View_SelectComponent_16(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),10,'li',[['role',
      'menuitem']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),7,'div',[['class','ui-select-choices-row']],[[2,'active',(null as any)]],
          [[(null as any),'mouseenter'],[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('mouseenter' === en)) {
              const pd_0:any = ((<any>_co.selectActive(_v.context.$implicit)) !== false);
              ad = (pd_0 && ad);
            }
            if (('click' === en)) {
              const pd_1:any = ((<any>_co.selectMatch(_v.context.$implicit,$event)) !== false);
              ad = (pd_1 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),4,'a',[['class','dropdown-item'],
          ['href','javascript:void(0)']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n              '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),[[8,'innerHTML',
          1]],(null as any),(null as any),(null as any),(null as any))),i0.ɵppd(2),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n          '])),(_l()(),i0.ɵted((null as any),['\n        ']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.isActive(_v.context.$implicit);
        _ck(_v,2,0,currVal_0);
        const currVal_1:any = _co.sanitize(i0.ɵunv(_v,6,0,_ck(_v,7,0,i0.ɵnov((<any>(<any>(<any>_v.parent).parent).parent),
            0),_v.context.$implicit.text,_co.inputValue)));
        _ck(_v,6,0,currVal_1);
      });
}
function View_SelectComponent_15(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'ul',[['class',
      'ui-select-choices dropdown-menu'],['role','menu']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_SelectComponent_16)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n      ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.options;
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
function View_SelectComponent_19(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'divider dropdown-divider']],(null as any),(null as any),(null as any),(null as any),
      (null as any)))],(null as any),(null as any));
}
function View_SelectComponent_20(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class',
      'ui-select-choices-row']],[[2,'active',(null as any)]],[[(null as any),'mouseenter'],
      [(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('mouseenter' === en)) {
      const pd_0:any = ((<any>_co.selectActive(_v.context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    if (('click' === en)) {
      const pd_1:any = ((<any>_co.selectMatch(_v.context.$implicit,$event)) !== false);
      ad = (pd_1 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵpod({'active':0}),(_l()(),i0.ɵted((null as any),
      ['\n            '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'a',[['class',
      'dropdown-item'],['href','javascript:void(0)']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n              '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'div',([] as any[]),[[8,'innerHTML',
          1]],(null as any),(null as any),(null as any),(null as any))),i0.ɵppd(2),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n          ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = 'ui-select-choices-row';
    const currVal_2:any = _ck(_v,2,0,_co.isActive(_v.context.$implicit));
    _ck(_v,1,0,currVal_1,currVal_2);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.isActive(_v.context.$implicit);
    _ck(_v,0,0,currVal_0);
    const currVal_3:any = _co.sanitize(i0.ɵunv(_v,6,0,_ck(_v,7,0,i0.ɵnov((<any>(<any>(<any>(<any>_v.parent).parent).parent).parent),
        0),_v.context.$implicit.text,_co.inputValue)));
    _ck(_v,6,0,currVal_3);
  });
}
function View_SelectComponent_18(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),10,'li',[['role',
      'menuitem']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_SelectComponent_19)),i0.ɵdid(16384,(null as any),
          0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'div',[['class','dropdown-header']],(null as any),(null as any),
          (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['',''])),(_l()(),i0.ɵted((null as any),['\n  \n          '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_SelectComponent_20)),i0.ɵdid(802816,
          (null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
          {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        ']))],
      (_ck,_v) => {
        const currVal_0:any = (_v.context.index > 0);
        _ck(_v,3,0,currVal_0);
        const currVal_2:any = _v.context.$implicit.children;
        _ck(_v,9,0,currVal_2);
      },(_ck,_v) => {
        const currVal_1:any = _v.context.$implicit.text;
        _ck(_v,6,0,currVal_1);
      });
}
function View_SelectComponent_17(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'ul',[['class',
      'ui-select-choices dropdown-menu'],['role','menu']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_SelectComponent_18)),
      i0.ɵdid(802816,(null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n      ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.options;
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
function View_SelectComponent_13(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),21,'div',[['class',
      'ui-select-container ui-select-multiple dropdown form-control open'],['tabindex',
      '0']],(null as any),[[(null as any),'keyup'],[(null as any),'focus'],[(null as any),
      'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>i0.ɵnov(_v,1).onClick($event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('keyup' === en)) {
      const pd_1:any = ((<any>_co.mainClick($event)) !== false);
      ad = (pd_1 && ad);
    }
    if (('focus' === en)) {
      const pd_2:any = ((<any>_co.focusToInput('')) !== false);
      ad = (pd_2 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(212992,(null as any),0,i2.OffClickDirective,
      ([] as any[]),{offClickHandler:[0,'offClickHandler']},(null as any)),(_l()(),
      i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      2,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),
      (null as any))),i0.ɵdid(278528,(null as any),0,i1.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{ngClass:[0,'ngClass']},(null as any)),
      i0.ɵpod({'ui-disabled':0}),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),4,'span',[['class','ui-select-match']],
              (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_SelectComponent_14)),i0.ɵdid(802816,(null as any),
          0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
              'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'input',[['autocapitalize','off'],['autocomplete','false'],['autocorrect',
              'off'],['class','form-control ui-select-search'],['role','combobox'],
              ['spellcheck','false'],['type','text']],[[8,'disabled',0],[8,'placeholder',
              0]],[[(null as any),'keydown'],[(null as any),'keyup'],[(null as any),
              'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:any = _v.component;
            if (('keydown' === en)) {
              const pd_0:any = ((<any>_co.inputEvent($event)) !== false);
              ad = (pd_0 && ad);
            }
            if (('keyup' === en)) {
              const pd_1:any = ((<any>_co.inputEvent($event,true)) !== false);
              ad = (pd_1 && ad);
            }
            if (('click' === en)) {
              const pd_2:any = ((<any>_co.matchClick($event)) !== false);
              ad = (pd_2 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n     '])),
      (_l()(),i0.ɵted((null as any),['\n     '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_SelectComponent_15)),i0.ɵdid(16384,(null as any),
          0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n  \n      '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_SelectComponent_17)),i0.ɵdid(16384,(null as any),
          0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.clickedOutside;
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _ck(_v,5,0,_co.disabled);
    _ck(_v,4,0,currVal_1);
    const currVal_2:any = _co.active;
    _ck(_v,10,0,currVal_2);
    const currVal_5:any = (((_co.optionsOpened && _co.options) && (_co.options.length > 0)) && !_co.firstItemHasChildren);
    _ck(_v,17,0,currVal_5);
    const currVal_6:any = (((_co.optionsOpened && _co.options) && (_co.options.length > 0)) && _co.firstItemHasChildren);
    _ck(_v,20,0,currVal_6);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_3:any = _co.disabled;
    const currVal_4:any = i0.ɵinlineInterpolate(1,'',((_co.active.length <= 0)? _co.placeholder: ''),
        '');
    _ck(_v,13,0,currVal_3,currVal_4);
  });
}
export function View_SelectComponent_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵpid(0,i3.HighlightPipe,([] as any[])),(_l()(),i0.ɵted((null as any),
      ['\n  '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_SelectComponent_1)),i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n\n  '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
      View_SelectComponent_13)),i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n  ']))],(_ck,_v) => {
    var _co:i4.SelectComponent = _v.component;
    const currVal_0:any = (_co.multiple === false);
    _ck(_v,3,0,currVal_0);
    const currVal_1:any = (_co.multiple === true);
    _ck(_v,6,0,currVal_1);
  },(null as any));
}
export function View_SelectComponent_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'ng-select',([] as any[]),
      (null as any),(null as any),(null as any),View_SelectComponent_0,RenderType_SelectComponent)),
      i0.ɵprd(5120,(null as any),i5.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i4.SelectComponent]),i0.ɵdid(114688,(null as any),0,i4.SelectComponent,[i0.ElementRef,
          i6.DomSanitizer],(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,2,0);
  },(null as any));
}
export const SelectComponentNgFactory:i0.ComponentFactory<i4.SelectComponent> = i0.ɵccf('ng-select',
    i4.SelectComponent,View_SelectComponent_Host_0,{allowClear:'allowClear',placeholder:'placeholder',
        idField:'idField',textField:'textField',childrenField:'childrenField',multiple:'multiple',
        items:'items',disabled:'disabled',active:'active'},{data:'data',selected:'selected',
        removed:'removed',typed:'typed',opened:'opened'},([] as any[]));
