// ngx-chips/dist/modules/components/dropdown/tag-input-dropdown.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '../../../../../ng2-material-dropdown/dist/src/modules/components/menu-item/ng2-menu-item.ngfactory';
import * as i3 from 'ng2-material-dropdown/dist/src/modules/components/menu-item/ng2-menu-item';
import * as i4 from 'ng2-material-dropdown/dist/src/modules/services/dropdown-state.service';
import * as i5 from 'ngx-chips/dist/modules/core/pipes/highlight.pipe';
import * as i6 from '../../../../../ng2-material-dropdown/dist/src/modules/components/dropdown/ng2-dropdown.ngfactory';
import * as i7 from 'ng2-material-dropdown/dist/src/modules/components/dropdown/ng2-dropdown';
import * as i8 from '../../../../../ng2-material-dropdown/dist/src/modules/components/menu/ng2-dropdown-menu.ngfactory';
import * as i9 from 'ng2-material-dropdown/dist/src/modules/components/menu/ng2-dropdown-menu';
import * as i10 from 'ngx-chips/dist/modules/components/dropdown/tag-input-dropdown.component';
import * as i11 from 'ngx-chips/dist/modules/components/tag-input/tag-input';
const styles_TagInputDropdown:any[] = ([] as any[]);
export const RenderType_TagInputDropdown:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_TagInputDropdown,data:{}});
function View_TagInputDropdown_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'span',([] as any[]),
      [[8,'innerHTML',1]],(null as any),(null as any),(null as any),(null as any))),
      i0.ɵppd(2),(_l()(),i0.ɵted((null as any),['\n            ']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = i0.ɵunv(_v,0,0,_ck(_v,1,0,i0.ɵnov((<any>(<any>_v.parent).parent),
            0),(<any>_v.parent).context.$implicit[_co.displayBy],_co.tagInput.inputForm.value.value));
        _ck(_v,0,0,currVal_0);
      });
}
function View_TagInputDropdown_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n            ']))],(null as any),
      (null as any));
}
function View_TagInputDropdown_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(16777216,(null as any),(null as any),2,(null as any),
      View_TagInputDropdown_4)),i0.ɵdid(540672,(null as any),0,i1.NgTemplateOutlet,
      [i0.ViewContainerRef],{ngTemplateOutlet:[0,'ngTemplateOutlet'],ngOutletContext:[1,
          'ngOutletContext']},(null as any)),i0.ɵpod({item:0,index:1,last:2}),(_l()(),
      i0.ɵand(0,(null as any),(null as any),0))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.templates.first;
    const currVal_1:any = _ck(_v,2,0,(<any>_v.parent).context.$implicit,(<any>_v.parent).context.index,
        (<any>_v.parent).context.last);
    _ck(_v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_TagInputDropdown_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'ng2-menu-item',
      ([] as any[]),(null as any),(null as any),(null as any),i2.View_Ng2MenuItem_0,
      i2.RenderType_Ng2MenuItem)),i0.ɵdid(16384,(null as any),0,i1.NgSwitch,([] as any[]),
      {ngSwitch:[0,'ngSwitch']},(null as any)),i0.ɵdid(49152,[[4,4]],0,i3.Ng2MenuItem,
      [i4.DropdownStateService,i0.ElementRef,i0.Renderer],{value:[0,'value']},(null as any)),
      (_l()(),i0.ɵted(0,['\n\n            '])),(_l()(),i0.ɵand(16777216,(null as any),
          0,1,(null as any),View_TagInputDropdown_2)),i0.ɵdid(278528,(null as any),
          0,i1.NgSwitchCase,[i0.ViewContainerRef,i0.TemplateRef,i1.NgSwitch],{ngSwitchCase:[0,
              'ngSwitchCase']},(null as any)),(_l()(),i0.ɵted(0,['\n\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),0,1,(null as any),View_TagInputDropdown_3)),
      i0.ɵdid(16384,(null as any),0,i1.NgSwitchDefault,[i0.ViewContainerRef,i0.TemplateRef,
          i1.NgSwitch],(null as any),(null as any)),(_l()(),i0.ɵted(0,['\n        ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:boolean = !!_co.templates.length;
        _ck(_v,1,0,currVal_0);
        const currVal_1:any = _v.context.$implicit;
        _ck(_v,2,0,currVal_1);
        const currVal_2:any = false;
        _ck(_v,5,0,currVal_2);
      },(null as any));
}
export function View_TagInputDropdown_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵpid(0,i5.HighlightPipe,([] as any[])),i0.ɵqud(402653184,1,
      {dropdown:0}),(_l()(),i0.ɵeld(0,(null as any),(null as any),13,'ng2-dropdown',
      ([] as any[]),(null as any),[['window','scroll']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('window:scroll' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,4).scrollListener()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },i6.View_Ng2Dropdown_0,i6.RenderType_Ng2Dropdown)),i0.ɵprd(512,(null as any),
      i4.DropdownStateService,i4.DropdownStateService,([] as any[])),i0.ɵdid(114688,
      [[1,4]],2,i7.Ng2Dropdown,[i4.DropdownStateService],(null as any),(null as any)),
      i0.ɵqud(335544320,2,{button:0}),i0.ɵqud(335544320,3,{menu:0}),(_l()(),i0.ɵted((null as any),
          ['\n    '])),(_l()(),i0.ɵeld(0,(null as any),1,6,'ng2-dropdown-menu',([] as any[]),
          (null as any),(null as any),(null as any),i8.View_Ng2DropdownMenu_0,i8.RenderType_Ng2DropdownMenu)),
      i0.ɵdid(507904,[[3,4]],1,i9.Ng2DropdownMenu,[i4.DropdownStateService,i0.ElementRef,
          i0.Renderer],{focusFirstElement:[0,'focusFirstElement'],offset:[1,'offset'],
          appendToBody:[2,'appendToBody']},(null as any)),i0.ɵqud(603979776,4,{items:1}),
      (_l()(),i0.ɵted(0,['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),0,
          1,(null as any),View_TagInputDropdown_1)),i0.ɵdid(802816,(null as any),0,
          i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
              'ngForOf']},(null as any)),(_l()(),i0.ɵted(0,['\n    '])),(_l()(),i0.ɵted((null as any),
          ['\n'])),(_l()(),i0.ɵted((null as any),['\n']))],(_ck,_v) => {
    var _co:i10.TagInputDropdown = _v.component;
    _ck(_v,4,0);
    const currVal_0:any = _co.focusFirstElement;
    const currVal_1:any = _co.offset;
    const currVal_2:any = _co.appendToBody;
    _ck(_v,9,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = _co.items;
    _ck(_v,13,0,currVal_3);
  },(null as any));
}
export function View_TagInputDropdown_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'tag-input-dropdown',
      ([] as any[]),(null as any),[['window','scroll']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('window:scroll' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1).scrollListener()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_TagInputDropdown_0,RenderType_TagInputDropdown)),i0.ɵdid(114688,(null as any),
      1,i10.TagInputDropdown,[i11.TagInputComponent],(null as any),(null as any)),
      i0.ɵqud(603979776,1,{templates:1})],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const TagInputDropdownNgFactory:i0.ComponentFactory<i10.TagInputDropdown> = i0.ɵccf('tag-input-dropdown',
    i10.TagInputDropdown,View_TagInputDropdown_Host_0,{offset:'offset',focusFirstElement:'focusFirstElement',
        showDropdownIfEmpty:'showDropdownIfEmpty',autocompleteObservable:'autocompleteObservable',
        minimumTextLength:'minimumTextLength',limitItemsTo:'limitItemsTo',displayBy:'displayBy',
        identifyBy:'identifyBy',matchingFn:'matchingFn',appendToBody:'appendToBody',
        autocompleteItems:'autocompleteItems'},{},([] as any[]));
