// ng-formly/lib/src/ui-bootstrap/types/select.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/forms';
import * as i2 from '@angular/common';
import * as i3 from 'ng-formly/lib/src/core/components/formly.attributes';
import * as i4 from 'ng-formly/lib/src/ui-bootstrap/types/select';
const styles_FormlyFieldSelect:any[] = ([] as any[]);
export const RenderType_FormlyFieldSelect:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_FormlyFieldSelect,data:{}});
function View_FormlyFieldSelect_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'option',[['value',
      '']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(147456,(null as any),0,i1.NgSelectOption,[i0.ElementRef,i0.Renderer,
          [2,i1.SelectControlValueAccessor]],{value:[0,'value']},(null as any)),i0.ɵdid(147456,
          (null as any),0,i1.ɵq,[i0.ElementRef,i0.Renderer,[8,(null as any)]],{value:[0,
              'value']},(null as any)),(_l()(),i0.ɵted((null as any),['','']))],(_ck,
      _v) => {
    const currVal_0:any = '';
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = '';
    _ck(_v,2,0,currVal_1);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_2:any = _co.to.placeholder;
    _ck(_v,3,0,currVal_2);
  });
}
function View_FormlyFieldSelect_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'option',([] as any[]),
      [[8,'disabled',0]],(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(147456,(null as any),0,i1.NgSelectOption,[i0.ElementRef,i0.Renderer,
          [2,i1.SelectControlValueAccessor]],{value:[0,'value']},(null as any)),i0.ɵdid(147456,
          (null as any),0,i1.ɵq,[i0.ElementRef,i0.Renderer,[8,(null as any)]],{value:[0,
              'value']},(null as any)),(_l()(),i0.ɵted((null as any),['\n           ',
          '\n         ']))],(_ck,_v) => {
    const currVal_1:any = _v.context.$implicit.value;
    _ck(_v,1,0,currVal_1);
    const currVal_2:any = _v.context.$implicit.value;
    _ck(_v,2,0,currVal_2);
  },(_ck,_v) => {
    const currVal_0:any = (<any>(<any>_v.parent).parent).context.$implicit.disabled;
    _ck(_v,0,0,currVal_0);
    const currVal_3:any = _v.context.$implicit.label;
    _ck(_v,3,0,currVal_3);
  });
}
function View_FormlyFieldSelect_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'optgroup',([] as any[]),
      [[8,'label',0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n         '])),(_l()(),i0.ɵand(16777216,(null as any),
      (null as any),1,(null as any),View_FormlyFieldSelect_4)),i0.ɵdid(802816,(null as any),
      0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,
          'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n       ']))],
      (_ck,_v) => {
        const currVal_1:any = (<any>_v.parent).context.$implicit.group;
        _ck(_v,3,0,currVal_1);
      },(_ck,_v) => {
        const currVal_0:any = i0.ɵinlineInterpolate(1,'',(<any>_v.parent).context.$implicit.label,
            '');
        _ck(_v,0,0,currVal_0);
      });
}
function View_FormlyFieldSelect_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'option',([] as any[]),
      [[8,'disabled',0]],(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(147456,(null as any),0,i1.NgSelectOption,[i0.ElementRef,i0.Renderer,
          [2,i1.SelectControlValueAccessor]],{value:[0,'value']},(null as any)),i0.ɵdid(147456,
          (null as any),0,i1.ɵq,[i0.ElementRef,i0.Renderer,[8,(null as any)]],{value:[0,
              'value']},(null as any)),(_l()(),i0.ɵted((null as any),['','']))],(_ck,
      _v) => {
    const currVal_1:any = (<any>_v.parent).context.$implicit.value;
    _ck(_v,1,0,currVal_1);
    const currVal_2:any = (<any>_v.parent).context.$implicit.value;
    _ck(_v,2,0,currVal_2);
  },(_ck,_v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.disabled;
    _ck(_v,0,0,currVal_0);
    const currVal_3:any = (<any>_v.parent).context.$implicit.label;
    _ck(_v,3,0,currVal_3);
  });
}
function View_FormlyFieldSelect_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,(null as any),
      (null as any),(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n       '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_FormlyFieldSelect_3)),i0.ɵdid(16384,(null as any),
          0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n       '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_FormlyFieldSelect_5)),i0.ɵdid(16384,(null as any),
          0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n      ']))],(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit.group;
    _ck(_v,3,0,currVal_0);
    const currVal_1:boolean = !_v.context.$implicit.group;
    _ck(_v,6,0,currVal_1);
  },(null as any));
}
export function View_FormlyFieldSelect_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),13,'select',[['class','form-control']],[[2,'ng-untouched',(null as any)],
          [2,'ng-touched',(null as any)],[2,'ng-pristine',(null as any)],[2,'ng-dirty',
              (null as any)],[2,'ng-valid',(null as any)],[2,'ng-invalid',(null as any)],
          [2,'ng-pending',(null as any)]],[[(null as any),'change'],[(null as any),
          'blur'],[(null as any),'focus']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('change' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,2).onChange($event.target.value)) !== false);
          ad = (pd_0 && ad);
        }
        if (('blur' === en)) {
          const pd_1:any = ((<any>i0.ɵnov(_v,2).onTouched()) !== false);
          ad = (pd_1 && ad);
        }
        if (('focus' === en)) {
          const pd_2:any = ((<any>i0.ɵnov(_v,7).onFocus()) !== false);
          ad = (pd_2 && ad);
        }
        if (('blur' === en)) {
          const pd_3:any = ((<any>i0.ɵnov(_v,7).onBlur()) !== false);
          ad = (pd_3 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i1.SelectControlValueAccessor,
      [i0.Renderer,i0.ElementRef],(null as any),(null as any)),i0.ɵprd(1024,(null as any),
      i1.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i1.SelectControlValueAccessor]),i0.ɵdid(540672,(null as any),0,i1.FormControlDirective,
      [[8,(null as any)],[8,(null as any)],[2,i1.NG_VALUE_ACCESSOR]],{form:[0,'form']},
      (null as any)),i0.ɵprd(2048,(null as any),i1.NgControl,(null as any),[i1.FormControlDirective]),
      i0.ɵdid(16384,(null as any),0,i1.NgControlStatus,[i1.NgControl],(null as any),
          (null as any)),i0.ɵdid(540672,(null as any),0,i3.FormlyAttributes,[i0.Renderer,
          i0.ElementRef],{field:[0,'field'],formControl:[1,'formControl']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_FormlyFieldSelect_1)),i0.ɵdid(16384,(null as any),
          0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_FormlyFieldSelect_2)),i0.ɵdid(802816,
          (null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
          {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵted((null as any),['\n  ']))],(_ck,_v) => {
    var _co:i4.FormlyFieldSelect = _v.component;
    const currVal_7:any = _co.formControl;
    _ck(_v,4,0,currVal_7);
    const currVal_8:any = _co.field;
    const currVal_9:any = _co.formControl;
    _ck(_v,7,0,currVal_8,currVal_9);
    const currVal_10:any = _co.to.placeholder;
    _ck(_v,10,0,currVal_10);
    const currVal_11:any = _co.selectOptions;
    _ck(_v,13,0,currVal_11);
  },(_ck,_v) => {
    const currVal_0:any = i0.ɵnov(_v,6).ngClassUntouched;
    const currVal_1:any = i0.ɵnov(_v,6).ngClassTouched;
    const currVal_2:any = i0.ɵnov(_v,6).ngClassPristine;
    const currVal_3:any = i0.ɵnov(_v,6).ngClassDirty;
    const currVal_4:any = i0.ɵnov(_v,6).ngClassValid;
    const currVal_5:any = i0.ɵnov(_v,6).ngClassInvalid;
    const currVal_6:any = i0.ɵnov(_v,6).ngClassPending;
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6);
  });
}
export function View_FormlyFieldSelect_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'formly-field-select',
      ([] as any[]),(null as any),(null as any),(null as any),View_FormlyFieldSelect_0,
      RenderType_FormlyFieldSelect)),i0.ɵdid(16760832,(null as any),0,i4.FormlyFieldSelect,
      ([] as any[]),(null as any),(null as any))],(_ck,_v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const FormlyFieldSelectNgFactory:i0.ComponentFactory<i4.FormlyFieldSelect> = i0.ɵccf('formly-field-select',
    i4.FormlyFieldSelect,View_FormlyFieldSelect_Host_0,{form:'form',field:'field',
        model:'model',options:'options'},{},([] as any[]));
