// ng-formly/lib/src/ui-bootstrap/wrappers/addons.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ng-formly/lib/src/ui-bootstrap/wrappers/addons';
const styles_FormlyWrapperAddons:any[] = ([] as any[]);
export const RenderType_FormlyWrapperAddons:i0.RendererType2 = i0.ɵcrt({encapsulation:2,
    styles:styles_FormlyWrapperAddons,data:{}});
function View_FormlyWrapperAddons_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'i',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),i0.ɵdid(278528,
      (null as any),0,i1.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,
          i0.Renderer],{ngClass:[0,'ngClass']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.to.addonLeft.class;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_FormlyWrapperAddons_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.to.addonLeft.text;
    _ck(_v,1,0,currVal_0);
  });
}
function View_FormlyWrapperAddons_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class',
      'input-group-addon']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.addonLeftClick($event)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgStyle,[i0.KeyValueDiffers,
      i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),i0.ɵpod({cursor:0}),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_FormlyWrapperAddons_2)),i0.ɵdid(16384,
          (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_FormlyWrapperAddons_3)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _ck(_v,2,0,(_co.to.addonLeft.onClick? 'pointer': 'inherit'));
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _co.to.addonLeft.class;
    _ck(_v,5,0,currVal_1);
    const currVal_2:any = _co.to.addonLeft.text;
    _ck(_v,8,0,currVal_2);
  },(null as any));
}
function View_FormlyWrapperAddons_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'i',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),i0.ɵdid(278528,
      (null as any),0,i1.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,
          i0.Renderer],{ngClass:[0,'ngClass']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.to.addonRight.class;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_FormlyWrapperAddons_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['','']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.to.addonRight.text;
    _ck(_v,1,0,currVal_0);
  });
}
function View_FormlyWrapperAddons_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class',
      'input-group-addon']],(null as any),[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.addonRightClick($event)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i1.NgStyle,[i0.KeyValueDiffers,
      i0.ElementRef,i0.Renderer],{ngStyle:[0,'ngStyle']},(null as any)),i0.ɵpod({cursor:0}),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_FormlyWrapperAddons_5)),i0.ɵdid(16384,
          (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_FormlyWrapperAddons_6)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n    ']))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_0:any = _ck(_v,2,0,(_co.to.addonRight.onClick? 'pointer': 'inherit'));
    _ck(_v,1,0,currVal_0);
    const currVal_1:any = _co.to.addonRight.class;
    _ck(_v,5,0,currVal_1);
    const currVal_2:any = _co.to.addonRight.text;
    _ck(_v,8,0,currVal_2);
  },(null as any));
}
export function View_FormlyWrapperAddons_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵqud(402653184,1,{fieldComponent:0}),(_l()(),i0.ɵted((null as any),
      ['\n    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class',
      'input-group']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_FormlyWrapperAddons_1)),i0.ɵdid(16384,
          (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(16777216,
          [[1,3],['fieldComponent',1]],(null as any),0,(null as any),(null as any),
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_FormlyWrapperAddons_4)),i0.ɵdid(16384,
          (null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n'])),(_l()(),i0.ɵted((null as any),
          ['\n  ']))],(_ck,_v) => {
    var _co:i2.FormlyWrapperAddons = _v.component;
    const currVal_0:any = _co.to.addonLeft;
    _ck(_v,5,0,currVal_0);
    const currVal_1:any = _co.to.addonRight;
    _ck(_v,10,0,currVal_1);
  },(null as any));
}
export function View_FormlyWrapperAddons_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'formly-wrapper-addons',
      ([] as any[]),(null as any),(null as any),(null as any),View_FormlyWrapperAddons_0,
      RenderType_FormlyWrapperAddons)),i0.ɵdid(49152,(null as any),0,i2.FormlyWrapperAddons,
      ([] as any[]),(null as any),(null as any))],(null as any),(null as any));
}
export const FormlyWrapperAddonsNgFactory:i0.ComponentFactory<i2.FormlyWrapperAddons> = i0.ɵccf('formly-wrapper-addons',
    i2.FormlyWrapperAddons,View_FormlyWrapperAddons_Host_0,{form:'form',field:'field',
        model:'model',options:'options'},{},([] as any[]));
