// ngx-chips/dist/modules/components/tag-input/tag-input.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from './tag-input.style.scss.shim.ngstyle';
import * as i1 from '@angular/core';
import * as i2 from '../tag/tag.component.ngfactory';
import * as i3 from 'ngx-chips/dist/modules/components/tag/tag.component';
import * as i4 from '@angular/common';
import * as i5 from 'ngx-chips/dist/modules/components/tag-input/tag-input';
import * as i6 from '../tag-input-form/tag-input-form.component.ngfactory';
import * as i7 from 'ngx-chips/dist/modules/components/tag-input-form/tag-input-form.component';
import * as i8 from '@angular/forms';
import * as i9 from 'ngx-chips/dist/modules/core/providers/drag-provider';
const styles_TagInputComponent:any[] = [i0.styles];
export const RenderType_TagInputComponent:i1.RendererType2 = i1.ɵcrt({encapsulation:0,
    styles:styles_TagInputComponent,data:{'animation':[{type:7,name:'animation',definitions:[{type:0,
        name:'in',styles:{type:6,styles:{opacity:1},offset:(null as any)}},{type:0,
        name:'out',styles:{type:6,styles:{opacity:0},offset:(null as any)}},{type:1,
        expr:':enter',animation:[{type:4,styles:{type:5,steps:[{type:6,styles:{opacity:0,
            offset:0,transform:'translate(0px, 20px)'},offset:(null as any)},{type:6,
            styles:{opacity:0.3,offset:0.3,transform:'translate(0px, -10px)'},offset:(null as any)},
            {type:6,styles:{opacity:0.5,offset:0.5,transform:'translate(0px, 0px)'},
                offset:(null as any)},{type:6,styles:{opacity:0.75,offset:0.75,transform:'translate(0px, 5px)'},
                offset:(null as any)},{type:6,styles:{opacity:1,offset:1,transform:'translate(0px, 0px)'},
                offset:(null as any)}]},timings:'{{ enter }}'}],options:(null as any)},
        {type:1,expr:':leave',animation:[{type:4,styles:{type:5,steps:[{type:6,styles:{opacity:1,
            transform:'translateX(0)',offset:0},offset:(null as any)},{type:6,styles:{opacity:1,
            transform:'translateX(-15px)',offset:0.7},offset:(null as any)},{type:6,
            styles:{opacity:0,transform:'translateX(100%)',offset:1},offset:(null as any)}]},
            timings:'{{ leave }}'}],options:(null as any)}],options:{}}]}});
function View_TagInputComponent_1(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),2,'tag',([] as any[]),
      [[8,'draggable',0],[1,'tabindex',0],[24,'@animation',0],[8,'draggable',0],[2,
          'moving',(null as any)]],[[(null as any),'onSelect'],[(null as any),'onRemove'],
          [(null as any),'onKeyDown'],[(null as any),'onTagEdited'],[(null as any),
              'onBlur'],[(null as any),'dragstart'],[(null as any),'drop'],[(null as any),
              'dragenter'],[(null as any),'dragover'],[(null as any),'dragleave'],
          [(null as any),'keydown']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('keydown' === en)) {
          const pd_0:any = ((<any>i1.ɵnov(_v,1).keydown($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('onSelect' === en)) {
          const pd_1:any = ((<any>_co.selectItem(_v.context.$implicit)) !== false);
          ad = (pd_1 && ad);
        }
        if (('onRemove' === en)) {
          const pd_2:any = ((<any>_co.onRemoveRequested(_v.context.$implicit,_v.context.index)) !== false);
          ad = (pd_2 && ad);
        }
        if (('onKeyDown' === en)) {
          const pd_3:any = ((<any>_co.handleKeydown($event,_v.context.$implicit)) !== false);
          ad = (pd_3 && ad);
        }
        if (('onTagEdited' === en)) {
          const pd_4:any = ((<any>_co.onTagEdited.emit($event)) !== false);
          ad = (pd_4 && ad);
        }
        if (('onBlur' === en)) {
          const pd_5:any = ((<any>_co.onTagBlurred($event,_v.context.index)) !== false);
          ad = (pd_5 && ad);
        }
        if (('dragstart' === en)) {
          const pd_6:any = ((<any>(_co.dragZone? _co.onDragStarted($event,_v.context.$implicit,
              _v.context.index): (undefined as any))) !== false);
          ad = (pd_6 && ad);
        }
        if (('drop' === en)) {
          const pd_7:any = ((<any>(_co.dragZone? _co.onTagDropped($event,_v.context.index): (undefined as any))) !== false);
          ad = (pd_7 && ad);
        }
        if (('dragenter' === en)) {
          const pd_8:any = ((<any>(_co.dragZone? _co.onDragOver($event): (undefined as any))) !== false);
          ad = (pd_8 && ad);
        }
        if (('dragover' === en)) {
          const pd_9:any = ((<any>(_co.dragZone? _co.onDragOver($event,_v.context.index): (undefined as any))) !== false);
          ad = (pd_9 && ad);
        }
        if (('dragleave' === en)) {
          const pd_10:any = ((<any>(_co.dragZone? _co.dragProvider.onDragEnd(): (undefined as any))) !== false);
          ad = (pd_10 && ad);
        }
        return ad;
      },i2.View_TagComponent_0,i2.RenderType_TagComponent)),i1.ɵdid(49152,[[2,4]],
      0,i3.TagComponent,[i1.ElementRef,i1.Renderer2,i1.ChangeDetectorRef],{model:[0,
          'model'],removable:[1,'removable'],editable:[2,'editable'],template:[3,'template'],
          displayBy:[4,'displayBy'],identifyBy:[5,'identifyBy'],index:[6,'index'],
          hasRipple:[7,'hasRipple'],disabled:[8,'disabled']},{onSelect:'onSelect',
          onRemove:'onRemove',onBlur:'onBlur',onKeyDown:'onKeyDown',onTagEdited:'onTagEdited'}),
      (_l()(),i1.ɵted((null as any),['\n        ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_5:any = _v.context.$implicit;
    const currVal_6:any = _co.removable;
    const currVal_7:any = _co.editable;
    const currVal_8:any = (!!_co.hasCustomTemplate()? _co.templates.first: (undefined as any));
    const currVal_9:any = _co.displayBy;
    const currVal_10:any = _co.identifyBy;
    const currVal_11:any = _v.context.index;
    const currVal_12:any = _co.ripple;
    const currVal_13:any = _co.disabled;
    _ck(_v,1,0,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,
        currVal_12,currVal_13);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i1.ɵinlineInterpolate(1,'',_co.editable,'');
    const currVal_1:any = 0;
    const currVal_2:any = _co.animationMetadata;
    const currVal_3:any = _co.dragZone;
    const currVal_4:any = i1.ɵnov(_v,1).moving;
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_TagInputComponent_2(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'progress-bar']],(null as any),(null as any),(null as any),(null as any),(null as any)))],
      (null as any),(null as any));
}
function View_TagInputComponent_4(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),4,'p',[['class',
      'error-message']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i1.ɵted((null as any),['\n        '])),(_l()(),i1.ɵeld(0,(null as any),
          (null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i1.ɵted((null as any),['',''])),(_l()(),
          i1.ɵted((null as any),['\n    ']))],(null as any),(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit;
    _ck(_v,3,0,currVal_0);
  });
}
function View_TagInputComponent_3(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),4,'div',([] as any[]),
      [[8,'className',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i1.ɵted((null as any),['\n    '])),(_l()(),i1.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_TagInputComponent_4)),i1.ɵdid(802816,
          (null as any),0,i4.NgForOf,[i1.ViewContainerRef,i1.TemplateRef,i1.IterableDiffers],
          {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i1.ɵted((null as any),['\n']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_1:any = _co.inputForm.getErrorMessages(_co.errorMessages);
        _ck(_v,3,0,currVal_1);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = i1.ɵinlineInterpolate(1,'error-messages ',(_co.theme || ''),
            '');
        _ck(_v,0,0,currVal_0);
      });
}
export function View_TagInputComponent_0(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[i1.ɵqud(402653184,1,{inputForm:0}),i1.ɵqud(671088640,2,{tags:1}),
      (_l()(),i1.ɵted((null as any),['\n\n'])),(_l()(),i1.ɵeld(0,(null as any),(null as any),
          16,'div',([] as any[]),[[1,'tabindex',0],[2,'ng2-tag-input--dropping',(null as any)],
              [2,'ng2-tag-input--disabled',(null as any)],[2,'ng2-tag-input--loading',
                  (null as any)],[2,'ng2-tag-input--invalid',(null as any)],[2,'ng2-tag-input--focused',
                  (null as any)]],[[(null as any),'click'],[(null as any),'drop'],
              [(null as any),'dragenter'],[(null as any),'dragover'],[(null as any),
                  'dragend']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i5.TagInputComponent = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.focus(true,false)) !== false);
              ad = (pd_0 && ad);
            }
            if (('drop' === en)) {
              const pd_1:any = ((<any>(_co.dragZone? _co.onTagDropped($event,(undefined as any)): (undefined as any))) !== false);
              ad = (pd_1 && ad);
            }
            if (('dragenter' === en)) {
              const pd_2:any = ((<any>(_co.dragZone? _co.onDragOver($event): (undefined as any))) !== false);
              ad = (pd_2 && ad);
            }
            if (('dragover' === en)) {
              const pd_3:any = ((<any>(_co.dragZone? _co.onDragOver($event): (undefined as any))) !== false);
              ad = (pd_3 && ad);
            }
            if (('dragend' === en)) {
              const pd_4:any = ((<any>(_co.dragZone? _co.dragProvider.onDragEnd(): (undefined as any))) !== false);
              ad = (pd_4 && ad);
            }
            return ad;
          },(null as any),(null as any))),i1.ɵdid(278528,(null as any),0,i4.NgClass,
          [i1.IterableDiffers,i1.KeyValueDiffers,i1.ElementRef,i1.Renderer],{ngClass:[0,
              'ngClass']},(null as any)),(_l()(),i1.ɵted((null as any),['\n\n    '])),
      (_l()(),i1.ɵted((null as any),['\n    '])),(_l()(),i1.ɵeld(0,(null as any),(null as any),
          8,'div',[['class','ng2-tags-container']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i1.ɵted((null as any),['\n        '])),
      (_l()(),i1.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TagInputComponent_1)),
      i1.ɵdid(802816,(null as any),0,i4.NgForOf,[i1.ViewContainerRef,i1.TemplateRef,
          i1.IterableDiffers],{ngForOf:[0,'ngForOf'],ngForTrackBy:[1,'ngForTrackBy']},
          (null as any)),(_l()(),i1.ɵted((null as any),['\n\n        '])),(_l()(),
          i1.ɵeld(0,(null as any),(null as any),2,'tag-input-form',([] as any[]),[[8,
              'hidden',0]],[[(null as any),'onSubmit'],[(null as any),'onBlur'],[(null as any),
              'click'],[(null as any),'onKeydown'],[(null as any),'onKeyup'],[(null as any),
              'inputTextChange']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i5.TagInputComponent = _v.component;
            if (('onSubmit' === en)) {
              const pd_0:any = ((<any>_co.onAddingRequested(false,_co.formValue)) !== false);
              ad = (pd_0 && ad);
            }
            if (('onBlur' === en)) {
              const pd_1:any = ((<any>_co.blur()) !== false);
              ad = (pd_1 && ad);
            }
            if (('click' === en)) {
              const pd_2:any = ((<any>(_co.dropdown? _co.dropdown.show(): (undefined as any))) !== false);
              ad = (pd_2 && ad);
            }
            if (('onKeydown' === en)) {
              const pd_3:any = ((<any>_co.fireEvents('keydown',$event)) !== false);
              ad = (pd_3 && ad);
            }
            if (('onKeyup' === en)) {
              const pd_4:any = ((<any>_co.fireEvents('keyup',$event)) !== false);
              ad = (pd_4 && ad);
            }
            if (('inputTextChange' === en)) {
              const pd_5:any = ((<any>(_co.inputText = $event)) !== false);
              ad = (pd_5 && ad);
            }
            return ad;
          },i6.View_TagInputForm_0,i6.RenderType_TagInputForm)),i1.ɵdid(114688,[[1,
          4]],0,i7.TagInputForm,([] as any[]),{placeholder:[0,'placeholder'],validators:[1,
          'validators'],asyncValidators:[2,'asyncValidators'],inputId:[3,'inputId'],
          inputClass:[4,'inputClass'],inputText:[5,'inputText'],tabindex:[6,'tabindex'],
          disabled:[7,'disabled']},{onSubmit:'onSubmit',onBlur:'onBlur',onKeyup:'onKeyup',
          onKeydown:'onKeydown',inputTextChange:'inputTextChange'}),(_l()(),i1.ɵted((null as any),
          ['\n        '])),(_l()(),i1.ɵted((null as any),['\n    '])),(_l()(),i1.ɵted((null as any),
          ['\n\n    '])),(_l()(),i1.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_TagInputComponent_2)),i1.ɵdid(16384,(null as any),0,i4.NgIf,[i1.ViewContainerRef,
          i1.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i1.ɵted((null as any),
          ['\n'])),(_l()(),i1.ɵted((null as any),['\n\n'])),(_l()(),i1.ɵted((null as any),
          ['\n'])),(_l()(),i1.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_TagInputComponent_3)),i1.ɵdid(16384,(null as any),0,i4.NgIf,[i1.ViewContainerRef,
          i1.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i1.ɵted((null as any),
          ['\n\n'])),i1.ɵncd((null as any),0),(_l()(),i1.ɵted((null as any),['\n']))],
      (_ck,_v) => {
        var _co:i5.TagInputComponent = _v.component;
        const currVal_6:any = i1.ɵinlineInterpolate(1,'ng2-tag-input ',(_co.theme || ''),
            '');
        _ck(_v,4,0,currVal_6);
        const currVal_7:any = _co.items;
        const currVal_8:any = _co.trackBy;
        _ck(_v,10,0,currVal_7,currVal_8);
        const currVal_10:any = (_co.items.length? _co.placeholder: _co.secondaryPlaceholder);
        const currVal_11:any = _co.validators;
        const currVal_12:any = _co.asyncValidators;
        const currVal_13:any = _co.inputId;
        const currVal_14:any = _co.inputClass;
        const currVal_15:any = _co.inputText;
        const currVal_16:any = _co.tabindex;
        const currVal_17:any = _co.disabled;
        _ck(_v,13,0,currVal_10,currVal_11,currVal_12,currVal_13,currVal_14,currVal_15,
            currVal_16,currVal_17);
        const currVal_18:any = _co.isLoading;
        _ck(_v,18,0,currVal_18);
        const currVal_19:any = _co.hasErrors();
        _ck(_v,23,0,currVal_19);
      },(_ck,_v) => {
        var _co:i5.TagInputComponent = _v.component;
        const currVal_0:any = (0 - 1);
        const currVal_1:any = _co.isDropping();
        const currVal_2:any = _co.disabled;
        const currVal_3:any = _co.isLoading;
        const currVal_4:any = _co.hasErrors();
        const currVal_5:any = _co.isInputFocused();
        _ck(_v,3,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5);
        const currVal_9:any = _co.maxItemsReached;
        _ck(_v,12,0,currVal_9);
      });
}
export function View_TagInputComponent_Host_0(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),4,'tag-input',([] as any[]),
      [[1,'tabindex',0]],(null as any),(null as any),View_TagInputComponent_0,RenderType_TagInputComponent)),
      i1.ɵprd(5120,(null as any),i8.NG_VALUE_ACCESSOR,(p0_0:any) => {
        return [p0_0];
      },[i5.TagInputComponent]),i1.ɵdid(4308992,(null as any),2,i5.TagInputComponent,
          [i1.Renderer2,i9.DragProvider],(null as any),(null as any)),i1.ɵqud(335544320,
          1,{dropdown:0}),i1.ɵqud(603979776,2,{templates:1})],(_ck,_v) => {
    _ck(_v,2,0);
  },(_ck,_v) => {
    const currVal_0:any = i1.ɵnov(_v,2).tabindexAttr;
    _ck(_v,0,0,currVal_0);
  });
}
export const TagInputComponentNgFactory:i1.ComponentFactory<i5.TagInputComponent> = i1.ɵccf('tag-input',
    i5.TagInputComponent,View_TagInputComponent_Host_0,{displayBy:'displayBy',identifyBy:'identifyBy',
        separatorKeys:'separatorKeys',separatorKeyCodes:'separatorKeyCodes',placeholder:'placeholder',
        secondaryPlaceholder:'secondaryPlaceholder',maxItems:'maxItems',validators:'validators',
        asyncValidators:'asyncValidators',onlyFromAutocomplete:'onlyFromAutocomplete',
        errorMessages:'errorMessages',theme:'theme',onTextChangeDebounce:'onTextChangeDebounce',
        inputId:'inputId',inputClass:'inputClass',clearOnBlur:'clearOnBlur',hideForm:'hideForm',
        addOnBlur:'addOnBlur',addOnPaste:'addOnPaste',pasteSplitPattern:'pasteSplitPattern',
        blinkIfDupe:'blinkIfDupe',removable:'removable',editable:'editable',allowDupes:'allowDupes',
        modelAsStrings:'modelAsStrings',trimTags:'trimTags',inputText:'inputText',
        ripple:'ripple',tabindex:'tabindex',disabled:'disabled',dragZone:'dragZone',
        onRemoving:'onRemoving',onAdding:'onAdding',animationDuration:'animationDuration'},
    {onAdd:'onAdd',onRemove:'onRemove',onSelect:'onSelect',onFocus:'onFocus',onBlur:'onBlur',
        onTextChange:'onTextChange',onPaste:'onPaste',onValidationError:'onValidationError',
        onTagEdited:'onTagEdited',inputTextChange:'inputTextChange'},['*']);
