// ngx-chips/dist/modules/components/tag/tag.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from './tag-component.style.scss.shim.ngstyle';
import * as i1 from '@angular/core';
import * as i2 from '@angular/common';
import * as i3 from '../icon/icon.ngfactory';
import * as i4 from 'ngx-chips/dist/modules/components/icon/icon';
import * as i5 from './tag-ripple.component.ngfactory';
import * as i6 from 'ngx-chips/dist/modules/components/tag/tag-ripple.component';
import * as i7 from 'ngx-chips/dist/modules/components/tag/tag.component';
const styles_TagComponent:any[] = [i0.styles];
export const RenderType_TagComponent:i1.RendererType2 = i1.ɵcrt({encapsulation:0,styles:styles_TagComponent,
    data:{}});
function View_TagComponent_2(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵted((null as any),['\n        ']))],(null as any),
      (null as any));
}
function View_TagComponent_1(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),
      [[1,'contenteditable',0]],(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i1.ɵted((null as any),['\n        '])),(_l()(),i1.ɵted((null as any),
          ['\n        '])),(_l()(),i1.ɵand(16777216,(null as any),(null as any),2,
          (null as any),View_TagComponent_2)),i1.ɵdid(540672,(null as any),0,i2.NgTemplateOutlet,
          [i1.ViewContainerRef],{ngTemplateOutlet:[0,'ngTemplateOutlet'],ngOutletContext:[1,
              'ngOutletContext']},(null as any)),i1.ɵpod({item:0,index:1}),(_l()(),
          i1.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = _co.template;
    const currVal_2:any = _ck(_v,5,0,_co.model,_co.index);
    _ck(_v,4,0,currVal_1,currVal_2);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.editing;
    _ck(_v,0,0,currVal_0);
  });
}
function View_TagComponent_4(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),2,'delete-icon',
      [['aria-label','Remove tag'],['role','button']],(null as any),[[(null as any),
          'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.remove($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },i3.View_DeleteIconComponent_0,i3.RenderType_DeleteIconComponent)),i1.ɵdid(49152,
      (null as any),0,i4.DeleteIconComponent,([] as any[]),(null as any),(null as any)),
      (_l()(),i1.ɵted((null as any),['\n        ']))],(null as any),(null as any));
}
function View_TagComponent_3(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),9,'div',[['class',
      'tag-wrapper']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i1.ɵted((null as any),['\n        '])),(_l()(),i1.ɵted((null as any),
          ['\n        '])),(_l()(),i1.ɵeld(0,(null as any),(null as any),1,'div',[['class',
          'tag__text inline'],['spellcheck','false']],[[1,'contenteditable',0],[1,
          'title',0]],[[(null as any),'keydown.enter'],[(null as any),'keydown.escape'],
          [(null as any),'click'],[(null as any),'blur']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('keydown.enter' === en)) {
          const pd_0:any = ((<any>_co.disableEditMode($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('keydown.escape' === en)) {
          const pd_1:any = ((<any>_co.disableEditMode($event)) !== false);
          ad = (pd_1 && ad);
        }
        if (('click' === en)) {
          const pd_2:any = ((<any>(_co.editing? $event.stopPropagation(): (undefined as any))) !== false);
          ad = (pd_2 && ad);
        }
        if (('blur' === en)) {
          const pd_3:any = ((<any>_co.onBlurred($event)) !== false);
          ad = (pd_3 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i1.ɵted((null as any),['\n            ',
          '\n        '])),(_l()(),i1.ɵted((null as any),['\n\n        '])),(_l()(),
          i1.ɵted((null as any),['\n        '])),(_l()(),i1.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_TagComponent_4)),i1.ɵdid(16384,(null as any),
          0,i2.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i1.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_3:any = _co.isDeleteIconVisible();
    _ck(_v,8,0,currVal_3);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.editing;
    const currVal_1:any = _co.getDisplayValue(_co.model);
    _ck(_v,3,0,currVal_0,currVal_1);
    const currVal_2:any = _co.getDisplayValue(_co.model);
    _ck(_v,4,0,currVal_2);
  });
}
function View_TagComponent_5(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),2,'tag-ripple',([] as any[]),
      [[1,'tabindex',0]],(null as any),(null as any),i5.View_TagRipple_0,i5.RenderType_TagRipple)),
      i1.ɵdid(49152,[[1,4]],0,i6.TagRipple,([] as any[]),{state:[0,'state']},(null as any)),
      (_l()(),i1.ɵted((null as any),['\n']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_1:any = _co.rippleState;
    _ck(_v,1,0,currVal_1);
  },(_ck,_v) => {
    const currVal_0:any = (0 - 1);
    _ck(_v,0,0,currVal_0);
  });
}
export function View_TagComponent_0(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[i1.ɵqud(671088640,1,{ripple:0}),(_l()(),i1.ɵeld(0,(null as any),
      (null as any),8,'div',([] as any[]),[[2,'disabled',(null as any)],[1,'tabindex',
          0],[1,'aria-label',0]],[[(null as any),'click'],[(null as any),'dblclick'],
          [(null as any),'mousedown'],[(null as any),'mouseup']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:i7.TagComponent = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>_co.select($event)) !== false);
          ad = (pd_0 && ad);
        }
        if (('dblclick' === en)) {
          const pd_1:any = ((<any>(_co.editable? _co.toggleEditMode(): (null as any))) !== false);
          ad = (pd_1 && ad);
        }
        if (('mousedown' === en)) {
          const pd_2:any = ((<any>(_co.rippleState = 'clicked')) !== false);
          ad = (pd_2 && ad);
        }
        if (('mouseup' === en)) {
          const pd_3:any = ((<any>(_co.rippleState = 'none')) !== false);
          ad = (pd_3 && ad);
        }
        return ad;
      },(null as any),(null as any))),i1.ɵdid(16384,(null as any),0,i2.NgSwitch,([] as any[]),
      {ngSwitch:[0,'ngSwitch']},(null as any)),(_l()(),i1.ɵted((null as any),['\n\n    '])),
      (_l()(),i1.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TagComponent_1)),
      i1.ɵdid(278528,(null as any),0,i2.NgSwitchCase,[i1.ViewContainerRef,i1.TemplateRef,
          i2.NgSwitch],{ngSwitchCase:[0,'ngSwitchCase']},(null as any)),(_l()(),i1.ɵted((null as any),
          ['\n\n    '])),(_l()(),i1.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_TagComponent_3)),i1.ɵdid(278528,(null as any),0,i2.NgSwitchCase,[i1.ViewContainerRef,
          i1.TemplateRef,i2.NgSwitch],{ngSwitchCase:[0,'ngSwitchCase']},(null as any)),
      (_l()(),i1.ɵted((null as any),['\n'])),(_l()(),i1.ɵted((null as any),['\n\n'])),
      (_l()(),i1.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TagComponent_5)),
      i1.ɵdid(16384,(null as any),0,i2.NgIf,[i1.ViewContainerRef,i1.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i1.ɵted((null as any),['\n']))],(_ck,_v) => {
    var _co:i7.TagComponent = _v.component;
    const currVal_3:boolean = !!_co.template;
    _ck(_v,2,0,currVal_3);
    const currVal_4:any = true;
    _ck(_v,5,0,currVal_4);
    const currVal_5:any = false;
    _ck(_v,8,0,currVal_5);
    const currVal_6:any = _co.isRippleVisible;
    _ck(_v,12,0,currVal_6);
  },(_ck,_v) => {
    var _co:i7.TagComponent = _v.component;
    const currVal_0:any = _co.disabled;
    const currVal_1:any = (0 - 1);
    const currVal_2:any = _co.getDisplayValue(_co.model);
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_TagComponent_Host_0(_l:any):i1.ɵViewDefinition {
  return i1.ɵvid(0,[(_l()(),i1.ɵeld(0,(null as any),(null as any),1,'tag',([] as any[]),
      [[2,'moving',(null as any)]],[[(null as any),'keydown']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('keydown' === en)) {
          const pd_0:any = ((<any>i1.ɵnov(_v,1).keydown($event)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },View_TagComponent_0,RenderType_TagComponent)),i1.ɵdid(49152,(null as any),
      0,i7.TagComponent,[i1.ElementRef,i1.Renderer2,i1.ChangeDetectorRef],(null as any),
      (null as any))],(null as any),(_ck,_v) => {
    const currVal_0:any = i1.ɵnov(_v,1).moving;
    _ck(_v,0,0,currVal_0);
  });
}
export const TagComponentNgFactory:i1.ComponentFactory<i7.TagComponent> = i1.ɵccf('tag',
    i7.TagComponent,View_TagComponent_Host_0,{model:'model',removable:'removable',
        editable:'editable',template:'template',displayBy:'displayBy',identifyBy:'identifyBy',
        index:'index',hasRipple:'hasRipple',disabled:'disabled'},{onSelect:'onSelect',
        onRemove:'onRemove',onBlur:'onBlur',onKeyDown:'onKeyDown',onTagEdited:'onTagEdited'},
    ([] as any[]));
